src/app/src/pages/Login.js

```diff
@@ -0,0 +1,91 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { Box, Button, Input, Text, VStack } from '@chakra-ui/react';
+
+import { URLS } from '../constants';
+import { login } from '../store/authSlice';
+
+// https://stackoverflow.com/questions/39254562/csrf-with-django-reactredux-using-axios
+axios.defaults.xsrfHeaderName = 'X-CSRFTOKEN';
+axios.defaults.xsrfCookieName = 'csrftoken';
+axios.defaults.withCredentials = true;
```

Comment:
Instead of re-initializing a copy of axios in every component we need it, consider following a pattern like this, as used in other projects:

```diff
diff --git a/src/app/src/api.js b/src/app/src/api.js
new file mode 100644
index 0000000..3d93549
--- /dev/null
+++ b/src/app/src/api.js
@@ -0,0 +1,10 @@
+import axios from 'axios';
+
+// Used to make authenticated HTTP requests to Django
+axios.defaults.xsrfHeaderName = 'X-CSRFToken';
+axios.defaults.xsrfCookieName = 'csrftoken';
+export const API = axios.create({
+    headers: {
+        credentials: 'same-origin',
+    },
+});
diff --git a/src/app/src/components/PrivateRoute.js b/src/app/src/components/PrivateRoute.js
index 58effc6..44d30ea 100644
--- a/src/app/src/components/PrivateRoute.js
+++ b/src/app/src/components/PrivateRoute.js
@@ -1,17 +1,11 @@
-import axios from 'axios';
-
 import { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { useLocation, useNavigate } from 'react-router-dom';
 
+import { API } from '../api';
 import { URLS } from '../constants';
 import { login, setLocationBeforeAuth } from '../store/authSlice';
 
-// https://stackoverflow.com/questions/39254562/csrf-with-django-reactredux-using-axios
-axios.defaults.xsrfHeaderName = 'X-CSRFTOKEN';
-axios.defaults.xsrfCookieName = 'csrftoken';
-axios.defaults.withCredentials = true;
-
 export default function PrivateRoute({ children }) {
     const signedIn = useSelector(state => state.auth.signedIn);
     const location = useLocation();
@@ -21,8 +15,7 @@ export default function PrivateRoute({ children }) {
     useEffect(() => {
         if (!signedIn) {
             dispatch(setLocationBeforeAuth(location));
-            axios
-                .get(URLS.LOGIN)
+            API.get(URLS.LOGIN)
                 .then(() => {
                     dispatch(login());
                 })
diff --git a/src/app/src/pages/Login.js b/src/app/src/pages/Login.js
index 5ef00a9..3a8053e 100644
--- a/src/app/src/pages/Login.js
+++ b/src/app/src/pages/Login.js
@@ -1,17 +1,12 @@
-import axios from 'axios';
 import { useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { useNavigate } from 'react-router-dom';
 import { Box, Button, Input, Text, VStack } from '@chakra-ui/react';
 
+import { API } from '../api';
 import { URLS } from '../constants';
 import { login } from '../store/authSlice';
 
-// https://stackoverflow.com/questions/39254562/csrf-with-django-reactredux-using-axios
-axios.defaults.xsrfHeaderName = 'X-CSRFTOKEN';
-axios.defaults.xsrfCookieName = 'csrftoken';
-axios.defaults.withCredentials = true;
-
 export default function Login() {
     const [email, setEmail] = useState('');
     const [password, setPassword] = useState('');
@@ -20,16 +15,15 @@ export default function Login() {
     const navigate = useNavigate();
 
     const loginRequest = () => {
-        axios
-            .post(URLS.LOGIN, {
-                email,
-                password,
-            })
+        API.post(URLS.LOGIN, {
+            email,
+            password,
+        })
             .then(() => {
                 dispatch(login());
             })
-            .catch(axiosError => {
-                setErrorDetail(axiosError.response.data?.detail);
+            .catch(apiError => {
+                setErrorDetail(apiError.response.data?.detail);
             });
     };
 
```

Reply:
Terrific. I'm going to raise a PR against the app template for this, as this would have saved me some research 🙏🏻 
d57d19181d4384202e2744c3bfd9470333327c0b